barretenberg/cpp/src/barretenberg/commitment_schemes/commitment_key.hpp

```diff
@@ -232,17 +232,19 @@ template <class Curve> class CommitmentKey {
 
         std::vector<Fr> scalars;
         scalars.reserve(total_num_scalars);
-        for (const auto& range : active_ranges) {
-            auto start = &polynomial[range.first];
-            auto end = &polynomial[range.second];
-            scalars.insert(scalars.end(), start, end);
-        }
         std::vector<G1> points;
         points.reserve(total_num_scalars * 2);
-        for (const auto& range : active_ranges) {
-            auto start = &point_table[2 * range.first];
-            auto end = &point_table[2 * range.second];
-            points.insert(points.end(), start, end);
+        for (const auto& [first, second] : active_ranges) {
+            {
+                auto start = &polynomial[first];
+                auto end = &polynomial[second];
+                scalars.insert(scalars.end(), start, end);
+            }
+            {
+                auto start = &point_table[2 * first];
+                auto end = &point_table[2 * second];
+                points.insert(points.end(), start, end);
+            }
```

Comment:
this is fine but probably better practice to just use different names to avoid the scoping weirdness, e.g. `pts_start`, `pts_end`. I'm a bit inconsistent on this bc I like to use this scoping hack in tests but for some reason not in production code - so do with this what you please!